File: ZoneCarotte.py
from sys import modules


class ZoneCarotte:
    upper_x: int
    upper_y: int
    lower_x: int
    lower_y: int
    w: int
    h: int
    
    def __init__(self, upper_x, upper_y, lower_x, lower_y, w, h):
        self.upper_x = upper_x
        self.upper_y = upper_y
        self.lower_x = lower_x
        self.lower_y = lower_y
        self.w = w
        self.h = h
        
    def getArea(self):
        return self.w * self.h

    def equals(self, other):
        if self.upper_x == other.upper_x and self.lower_x == other.lower_x and self.upper_y == other.upper_y and self.lower_y == other.lower_y:
            return True
        else:
            return False
        
    def existsInArray(self, array):
        exists = False
        for i in range(len(array)):
            if self.equals(array[i]):
                exists = True
        return exists

File: mavenproject1/src/main/java/com/mycompany/mavenproject1/ZoneCarotte.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.mavenproject1;

import java.util.List;

/**
 *
 * @author MSI
 */
public class ZoneCarotte {
    
    int upper_x;
    int upper_y;
    int lower_x;
    int lower_y;
    int w;
    int h;
    
    public ZoneCarotte(int upper_x,int upper_y,int lower_x,int lower_y,int w,int h){
        
        this.upper_x = upper_x;
        this.upper_y = upper_y;
        this.lower_x = lower_x;
        this.lower_y = lower_y;
        this.w = w;
        this.h = h;
   
    }
    
    public int getArea(){
        
        return this.w * this.h;
        
    }
    
    public boolean equals(ZoneCarotte other){

        if(this.upper_x == other.upper_x){
            
            return true;
            
        }else{
        
            return false;
            
        }
    }
    
    public boolean existsInArray(List<ZoneCarotte> array){
        
        boolean exists = false;

        for(int i = 0; i < array.size(); i++ ){
            
            if(this.equals(array.get(i))){
                
                exists = true;
            }
            
        }
        
        return exists;
        
    }
    
    
}
